examples/guided.py

```diff
@@ -212,14 +212,8 @@ def ask_user_questions():
 
 	# Ask about audio server selection if one is not already set
 	if not archinstall.arguments.get('audio', None):
-		# only ask for audio server selection on a desktop profile
-		if str(archinstall.arguments['profile']) == 'Profile(desktop)':
-			archinstall.arguments['audio'] = archinstall.ask_for_audio_selection()
-		else:
-			# packages installed by a profile may depend on audio and something may get installed anyways, not much we can do about that.
-			# we will not try to remove packages post-installation to not have audio, as that may cause multiple issues
-			archinstall.arguments['audio'] = None
-
+		archinstall.arguments['audio'] = archinstall.ask_for_audio_selection()
```

Comment:
So this might cause issues - since this was originally written for desktop, it'll prompt for audio on any install incl. minimal, and audio selection is an either/or for pulseaudio or pipewire right now. The rest of this change I'm fine with, but -1 on this unless we somehow fundamentally change audio selection since you'd have audio on a min. install this way no matter what.

Reply:
Perhaps we should ask for audio if the selected profile is a desktop profile OR the selected profile it is in the list of desktop profiles in desktop.py

Reply:
To keep that clean, we may need to implement a is_desktop_profile(profile_name) -> bool helper function or something

Reply:
This is just a prompt right, the user could leave this blank, but yeah +1 for the helper function

Reply:
That's a possibility, but I think that is might be better to just rework this logic so we display it for just desktop profiles for now, since those will have audio anyways. Part of the issue with 'none' in a DE is that packages will get pulled in by deps anyways and you'll end up with audio even if you select none. But pipewire is a 'replacement' for pulseaudio. Hence, it makes sense the way it is now.

Reply:
Maybe we can have (like we do with `top_level_profile`) a meta variable that specifies what the profile supports, for instance any desktop profile (or desktop itself) could have:
```
import archinstall

__supports__ = ['audio', 'desktop', 'xorg']
```
Or something similar, and we could create the helper function you mentioned dylan, but instead of just doing `is_desktop_profile` we could return the supported elements of the profile?

Reply:
All desktop profiles support audio, so I'm not sure if this makes sense

Reply:
Actually, audio can work without a DE at all if you're adventurous

Reply:
Absolutely, which is why profiles should define that they support `audio`, which could enable the question "Which audio-server do you want?". I just saw it as a way to trigger the question by using the meta-definitions of the profiles : )

Reply:
All profiles support audio. There's no value in specifying this in my opinion. Even on a server install you might want it if you're trying to use it for audio streaming or something like that.

Reply:
I thought about this a bit more and here's what I came up with:

Write the is_desktop_profile function
If it's a desktop, do a generic select of the set pipewire, pulseaudio. Otherwise, do a generic select of the set pipewire, pulseaudio, none. If it's already set via config, suppress the prompt.